contracts/connection-router/src/events.rs

```diff
@@ -104,8 +104,8 @@ impl From<ChainUnfrozen> for Event {
 impl From<NewMessage> for Vec<Attribute> {
     fn from(other: NewMessage) -> Self {
         vec![
-            ("id", other.id_on_chain).into(),
-            ("source_chain", other.source_chain).into(),
+            ("id", other.id.message_id).into(),
+            ("source_chain", other.id.chain).into(),
```

Comment:
I'd suggest merging these attributes into one as well. 
```suggestion
            ("id", other.id).into(),
```